packages/pointing-frame/src/LionPointingFrame.js

```diff
@@ -0,0 +1,280 @@
+    const pointingFrameWidth = this.getBoundingClientRect().width;
+
+    pointerStyles.display = 'block';
+    Object.assign(pointerEl.style, pointerStyles);
+    const pointerHeight = pointerEl.firstElementChild.getBoundingClientRect().height;
+    const pointerWidth = pointerEl.firstElementChild.getBoundingClientRect().width;
+
+    const { vertical, horizontal, horizontalIsPrimary } = getPlacement(this.position);
+
+    if (horizontalIsPrimary) {
+      // pointerHeight / 2 here is a correction to do with the 270 and 90 deg rotations
+      if (horizontal === 'left') {
+        pointerStyles.transform = 'rotate(90deg)';
+        pointerStyles.right = `${-(pointerHeight / 2)}px`;
+      } else if (horizontal === 'right') {
+        pointerStyles.transform = 'rotate(270deg)';
+        pointerStyles.left = `${-(pointerHeight / 2)}px`;
+      }
+
+      const spacing = this.invokerHeight / 2 - pointerWidth / 2 - this.pointerDistanceFromEdge;
+      if (vertical === 'top') {
+        if (spacing < 0) {
+          pointerStyles.bottom = `${pointerHeight / 2 + this.pointerDistanceFromEdge}px`;
+        } else {
+          pointerStyles.bottom = `${pointerHeight / 2 +
+            this.invokerHeight / 2 -
+            pointerWidth / 2}px`;
+        }
+      } else if (vertical === 'center') {
+        pointerStyles.bottom = `${pointingFrameHeight / 2 - pointerHeight / 2}px`;
+      } else if (vertical === 'bottom') {
+        if (spacing < 0) {
+          pointerStyles.top = `${pointerHeight / 2 + this.pointerDistanceFromEdge}px`;
+        } else {
+          pointerStyles.top = `${pointerHeight / 2 + this.invokerHeight / 2 - pointerWidth / 2}px`;
+        }
+      }
+    } else {
+      const spacing = this.invokerWidth / 2 - pointerWidth / 2 - this.pointerDistanceFromEdge;
+      if (horizontal === 'left') {
+        if (spacing < 0) {
+          pointerStyles.right = `${this.pointerDistanceFromEdge}px`;
+        } else {
+          pointerStyles.right = `${this.invokerWidth / 2 - pointerWidth / 2}px`;
+        }
+      } else if (horizontal === 'center') {
+        pointerStyles.bottom = 0;
+        pointerStyles.left = `${pointingFrameWidth / 2 - pointerWidth / 2}px`;
+      } else if (horizontal === 'right') {
+        if (spacing < 0) {
+          pointerStyles.left = `${this.pointerDistanceFromEdge}px`;
+        } else {
+          pointerStyles.left = `${this.invokerWidth / 2 - pointerWidth / 2}px`;
+        }
+      }
+
+      if (vertical === 'top') {
+        pointerStyles.transform = 'rotate(180deg)';
+        pointerStyles.bottom = 0;
+      } else if (vertical === 'bottom') {
+        pointerStyles.top = 0;
+      }
+    }
+
+    Object.assign(pointerEl.style, pointerStyles);
+  }
+
+  __setFrameStyles(pointerEl) {
+    const pointingFrameEl = this.shadowRoot.querySelector('.pointing-frame');
+
+    const frameStyles = {
+      boxSizing: 'border-box',
+      fontSize: '16px',
+      color: 'black',
+      backgroundColor: 'white',
```

Comment:
I think these are run in an update loop? The top 4 props mainly concern styling and setting them once (outside update loop) would be better